### ISSUE TITLE
Support for constraints-based CNF reduction [RFE]

### ISSUE THREAD

**Issue by woncount** [msoos/cryptominisat#97]
I have a simple compression/encryption routine that is called in a nested fashion (twice) per data set - cryptominisat  can successfully solve the algorithm (good  job!), however it is taking more time than necessary, presumably because it is traversing more conditions than necessary, because I know for sure that the input data is extremely restricted in each pass, i.e. because of the value types and value ranges passed to the routine.

The input is usually ~1kb, while the output of the first pass is only 512 bytes, and used as input in the second pass, valid bytes are also limited.

So, basically I am thinking in terms of adding constraints to inputs, in order to simplify the SAT process by restricting valid inputs/outputs to hopefully reduce the complexity of the CNF.
